packages/core/src/utils.ts

```diff
@@ -405,6 +405,7 @@ export function updateContext<TContext, TEvent extends EventObject>(
   assignActions: Array<AssignAction<TContext, TEvent>>,
   state?: State<TContext, TEvent>
 ): TContext {
+  warn(!!context, 'Attempting to update undefined context');
```

Comment:
Could you wrap this with env check, like it's done for example here?
https://github.com/davidkpiano/xstate/blob/5eec678629074989bcc45a63ff77dbbc12126775/packages/core/src/interpreter.ts#L220-L225

Reply:
Actually, `warn` already handles this: `warn` is [initialized](https://github.com/davidkpiano/xstate/blob/master/packages/core/src/utils.ts#L439) as `() => {}`, and is only overwritten when `IS_PRODUCTION` is false.

Reply:
Right, but the error message is still in the source code for production builds... I'll have to look into how other libraries like React handle this.

Reply:
React has dropped (or is in the process of dropping) those invariant/warn helpers in favor of no abstraction - just do the env-guarded if statement and throw an error if needed

Reply:
> Right, but the error message is still in the source code for production builds... I'll have to look into how other libraries like React handle this.

Makes sense. Addressed in 20f5235